relay-metrics/src/aggregation.rs

```diff
@@ -745,6 +745,46 @@ impl BucketKey {
         std::hash::Hash::hash(self, &mut hasher);
         hasher.finalize() as i64
     }
+
+    /// Validates the metric name and tags' keys and values fulfill the max length threshold.
+    ///
+    /// If the metric name surpasses the threshold, returns an Err. Tags with surpassing key or
+    /// value lengths are removed from the bucket.
+    fn validate_string_lengths(mut self, aggregator_config: &AggregatorConfig) -> Result<Self, ()> {
```

Comment:
Please move this function to be separate from the type, ideally static method on aggregator.